Updates wasm middleware documentation with compilation instructions (#2663)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-wasm.md

## Spec metadata fields

| Field    | Details                                        | Example        |
|----------|------------------------------------------------|----------------|
| path     | A relative or absolute path to the Wasm binary | "./hello.wasm" |

> | poolSize | The instance count of the Wasm binary          | 10             |

Reviewer: Why did you remove the `runtime` property from this table? Was this unintentional?

Reviewer: Also can you add the `poolsize` property into the YAML example.

Author: > Why did you remove the runtime property from this table? Was this unintentional?

This was intentional because the only other runtimes that could work are build flags, not runtime flags, as they would require CGO (and are build flags in wapc-go). The first impl only supports one runtime, wazero as there's no plan to optimizing towards CGO per the below thread.

https://github.com/dapr/components-contrib/pull/1833#issuecomment-1186051640

> Also can you add the poolsize property into the YAML example.

sure can do

> | poolSize | The instance count of the Wasm binary          | 10             |

Reviewer: Need to say whether `poolsize is optional or required`. And describe a little more on what this does. Does it instantiate this number of Wasm binary images?

Author: > Need to say whether poolsize is optional or required. And describe a little more on what this does. Does it instantiate this number of Wasm binary images?

I will add text that it is optional. WRT how to describe it, the pool size is the amount of instances of the binary, specified by path. instance and binary is to wasm as container and image is to docker. Advice on rewording?

Author: I pitched a rewording. Let me know if it needs more clarity

      type: middleware.http.wasm.basic
```

### Generating Wasm

This component allows you to rewrite a request URI with custom logic compiled
to a Wasm using the waPC protocol. The `rewrite` function receives the request
URI and returns an update as necessary.

To compile your Wasm, you must compile source using a waPC guest SDK such as
[TinyGo](https://github.com/wapc/wapc-guest-tinygo).

Here's an example in TinyGo:
```go
package main

import "github.com/wapc/wapc-guest-tinygo"

func main() {
	wapc.RegisterFunctions(wapc.Functions{"rewrite": rewrite})
}

// rewrite returns a new URI if necessary.
func rewrite(requestURI []byte) ([]byte, error) {
	if string(requestURI) == "/v1.0/hi" {
		return []byte("/v1.0/hello"), nil
	}
	return requestURI, nil
}
```

> If using TinyGo, compile like so and set the `path` attribute to the output:

Reviewer: ```suggestion
If using TinyGo, compile as shown below and set the `path` attribute to the output:
```
Also There is no `path` attribute below in the example

Author: > Also There is no path attribute below in the example

I'll reword to say "spec metadata field named path" unless you can figure a way to be unambiguous in a less chatty way.
